# Patch to check for GTest and Gflags in a visible way,
# add GTEST_BOTH_LIBRARIES for snappy-unittest
#
# Copyright (C) 2017 Georgi D. Sotirov <gdsotirov@dir.bg>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License
# as published by the Free Software Foundation; either version 2
# of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
#
# $Id: snappy-1.1.7.diff,v 1.1 2017/08/25 13:06:52 gsotirov Exp $
#
diff -urNad snappy-1.1.7-orig/CMakeLists.txt snappy-1.1.7/CMakeLists.txt
--- snappy-1.1.7-orig/CMakeLists.txt	2017-08-25 02:54:23.000000000 +0300
+++ snappy-1.1.7/CMakeLists.txt	2017-08-25 16:03:01.169417100 +0300
@@ -37,12 +37,12 @@
 check_symbol_exists("mmap" "sys/mman.h" HAVE_FUNC_MMAP)
 check_symbol_exists("sysconf" "unistd.h" HAVE_FUNC_SYSCONF)
 
-find_package(GTest QUIET)
+find_package(GTest)
 if(GTEST_FOUND)
   set(HAVE_GTEST 1)
 endif(GTEST_FOUND)
 
-find_package(Gflags QUIET)
+find_package(Gflags)
 if(GFLAGS_FOUND)
   set(HAVE_GFLAGS 1)
 endif(GFLAGS_FOUND)
@@ -118,7 +118,7 @@
       "${PROJECT_SOURCE_DIR}/snappy-test.cc"
   )
   target_compile_definitions(snappy_unittest PRIVATE -DHAVE_CONFIG_H)
-  target_link_libraries(snappy_unittest snappy ${GFLAGS_LIBRARIES})
+  target_link_libraries(snappy_unittest snappy ${GFLAGS_LIBRARIES} ${GTEST_BOTH_LIBRARIES})
 
   if(HAVE_LIBZ)
     target_link_libraries(snappy_unittest z)
